### DISCUSSION TITLE
How to build Angband on Debian

### DISCUSSION THREAD

**Discussion by shelches** [angband/angband#5729] [General]
Are these instructions still correct for building Angband on Debian Linux?

[https://angband.readthedocs.io/en/latest/hacking/compiling.html#linux-other-unix](https://angband.readthedocs.io/en/latest/hacking/compiling.html#linux-other-unix)

They work just fine for me (after installing some prerequisites, such as autotools-dev, automake, and libncurses-dev), and I've been running Angband (including the Borg) directly from the src directory in a terminal (no GUI).

However, I heard a rumor there's a better, or alternative, way that should be used to build Angband from source on Debian Linux.  Any insight or information appreciated.

Thanks!

**Comment by NickMcConnell** [comment]
That's the way I do it (on openSUSE).

**Reply by shelches** [reply]
OK - thanks for the sanity check, Nick!

**Comment by magnate** [comment]
I am hoping to package 4.2.5 for Debian, if I can re-learn its build system!

**Comment by magnate** [comment]
Nick, do I still need to install sphinx to build the manual?

**Reply by NickMcConnell** [reply]
I think so?  I just push it all to Github for now these days...

**Reply by magnate** [reply]
Ok thanks

**Comment by magnate** [comment]
Just FYI, I am getting a very basic error trying to build --with-no-install before I start adding the Debian packaging stuff: 

make[3]: *** No rule to make target '/usr/lib/gcc/x86_64-linux-gnu/11/include/stdbool.h', needed by 'cave.o'.  Stop.


This error is happening because my gcc is v12 so the /usr/lib/gcc/x86_64-linux-gnu/ directory does not contain a subdir 11/, only 12/ (and 10/ and others). It's possible that my system is misconfigured, but just sharing in case it affects anyone else, or anyone can help.

**Reply by magnate** [reply]
Hmm this went away, so obviously something very bizarre was happening on my system late last night. Sorry for the noise.

**Comment by magnate** [comment]
Debian packages are now available at http://bass.fugue.org/angband/4.2.5/

You need angband_4.2.5+dfsg-1_amd64.deb and angband-data_4.2.5+dfsg-1_all.deb
